### PR TITLE
Advancements

### PR DESCRIPTION
**Pull request by OroArmor**

### CODE REVIEW

File: mappings/net/minecraft/advancement/criterion/LocationCriterion.mapping [hunk 1]
```diff
@@ -0,0 +1,19 @@
+CLASS net/minecraft/unmapped/C_maikngon net/minecraft/advancement/criterion/LocationCriterion
+	FIELD f_qjamvpft id Lnet/minecraft/unmapped/C_ncpywfca;
+	METHOD <init> (Lnet/minecraft/unmapped/C_ncpywfca;)V
+		ARG 1 ud
```
Comment (triphora) [inline mappings/net/minecraft/advancement/criterion/LocationCriterion.mapping]:
```suggestion
		ARG 1 id
```

File: mappings/net/minecraft/advancement/criterion/LocationCriterion.mapping [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	METHOD m_jeouaxek (Lnet/minecraft/unmapped/C_maikngon$C_zsxskcab;)Z
+		ARG 0 conditions
+	METHOD m_qjisaduw trigger (Lnet/minecraft/unmapped/C_mxrobsgg;)V
+		ARG 1 player
+	CLASS C_zsxskcab Conditions
+		METHOD m_enfvbgsy createLocation (Lnet/minecraft/unmapped/C_ypvebgnl;)Lnet/minecraft/unmapped/C_maikngon$C_zsxskcab;
+			ARG 0 location
+		METHOD m_eyeihplq createWearingItemOnFeetStandingOnBlock (Lnet/minecraft/unmapped/C_mmxmpdoq;Lnet/minecraft/unmapped/C_vorddnax;)Lnet/minecraft/unmapped/C_maikngon$C_zsxskcab;
```
Comment (triphora) [inline mappings/net/minecraft/advancement/criterion/LocationCriterion.mapping]: `createWearingBootsStandingOnBlock`? That's a bad name too but this feels too long and unwieldy

File: mappings/net/minecraft/advancement/criterion/PickedUpThrownItemCriterion.mapping [hunk 3]
```diff
@@ -0,0 +1,30 @@
+CLASS net/minecraft/unmapped/C_mntpixlf net/minecraft/advancement/criterion/PickedUpThrownItemCriterion
+	FIELD f_opdeixgk id Lnet/minecraft/unmapped/C_ncpywfca;
+	METHOD <init> (Lnet/minecraft/unmapped/C_ncpywfca;)V
+		ARG 1 id
+	METHOD m_hsjbqbnl trigger (Lnet/minecraft/unmapped/C_mxrobsgg;Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_astfners;)V
+		ARG 1 player
+		ARG 2 stack
+		ARG 3 entity
+	METHOD m_ypucoayn (Lnet/minecraft/unmapped/C_mxrobsgg;Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_iakykpgh;Lnet/minecraft/unmapped/C_mntpixlf$C_oqopxfol;)Z
+		ARG 3 contitions
```
Comment (triphora) [inline mappings/net/minecraft/advancement/criterion/PickedUpThrownItemCriterion.mapping]:
```suggestion
		ARG 3 conditions
```

File: mappings/net/minecraft/advancement/criterion/PlayerInteractedWithEntityCriterion.mapping [hunk 4]
```diff
@@ -13,6 +13,13 @@ CLASS net/minecraft/unmapped/C_bvljsgzr net/minecraft/advancement/criterion/Play
...
 			ARG 2 item
 			ARG 3 entity
+		METHOD m_qisejrky create (Lnet/minecraft/unmapped/C_kvkvpjlm$C_tthiakui;Lnet/minecraft/unmapped/C_afravpde$C_qwujnjpk;Lnet/minecraft/unmapped/C_kvkvpjlm$C_tthiakui;)Lnet/minecraft/unmapped/C_bvljsgzr$C_dwldgxex;
+			ARG 0 player
+			ARG 1 item
+			ARG 2 entity
 		METHOD m_sxysgxiu test (Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_iakykpgh;)Z
 			ARG 1 stack
 			ARG 2 context
+		METHOD m_yqpynebn createNoPlayer (Lnet/minecraft/unmapped/C_afravpde$C_qwujnjpk;Lnet/minecraft/unmapped/C_kvkvpjlm$C_tthiakui;)Lnet/minecraft/unmapped/C_bvljsgzr$C_dwldgxex;
```
Comment (triphora) [inline mappings/net/minecraft/advancement/criterion/PlayerInteractedWithEntityCriterion.mapping]:
```suggestion
		METHOD m_yqpynebn createWithoutPlayer (Lnet/minecraft/unmapped/C_afravpde$C_qwujnjpk;Lnet/minecraft/unmapped/C_kvkvpjlm$C_tthiakui;)Lnet/minecraft/unmapped/C_bvljsgzr$C_dwldgxex;
```

File: mappings/net/minecraft/advancement/criterion/LocationCriterion.mapping (hunk 2 above)
Comment (OroArmor) [reply mappings/net/minecraft/advancement/criterion/LocationCriterion.mapping]: It's for checking the player has a specific item on their feet while also being on a block, probably used to the sneak 100 advancement. Tbh people shouldn't really call these but I couldn't think of a better name

**Review by triphora — APPROVED** [review]

File: mappings/net/minecraft/advancement/criterion/LocationCriterion.mapping (hunk 2 above)
Comment (NoComment1105) [reply mappings/net/minecraft/advancement/criterion/LocationCriterion.mapping]:
```suggestion
		METHOD m_eyeihplq checkFootwearAndBlock (Lnet/minecraft/unmapped/C_mmxmpdoq;Lnet/minecraft/unmapped/C_vorddnax;)Lnet/minecraft/unmapped/C_maikngon$C_zsxskcab;
``` 
perhaps?

**Review by NoComment1105 — APPROVED** [review]

**Review by IotaBread — APPROVED** [review]

**Pull request merged**